runtime/src/main/scala/com/ing/baker/runtime/actor/recipe_manager/RecipeManager.scala

```diff
@@ -17,50 +17,52 @@ object RecipeManager {
 
   //Events
   //When a recipe is added
-  case class RecipeAdded(recipeId: String, compiledRecipe: CompiledRecipe) extends BakerProtoMessage
+  case class RecipeAdded(recipeId: String, compiledRecipe: CompiledRecipe, timeStamp: Long) extends BakerProtoMessage
+
 }
 
 class RecipeManager extends PersistentActor with ActorLogging {
 
-  val compiledRecipes: mutable.Map[String, CompiledRecipe] = mutable.Map[String, CompiledRecipe]()
+  val compiledRecipes: mutable.Map[String, (CompiledRecipe, Long)] = mutable.Map[String, (CompiledRecipe, Long)]()
 
   private def hasCompiledRecipe(compiledRecipe: CompiledRecipe): Option[String] =
-    compiledRecipes.find(_._2 == compiledRecipe).map(_._1)
+    compiledRecipes.find(_._2._1 == compiledRecipe).map(_._1)
```

Comment:
This looks weird, all the underscores, maybe refactor to find `{ case ((_, (_, recipe)) => recipe == compiledRecipe }`

Reply:
Changed